lib/rubygems.rb

```diff
@@ -791,33 +792,30 @@ def self.write_binary(path, data)
   end
 
   ##
-  # Open a file with given flags. It requires special logic on Windows, like
-  # protecting access with flock
+  # Open a file with given flags
 
   def self.open_file(path, flags, &block)
-    if !java_platform? && win_platform?
-      open_file_with_flock(path, flags, &block)
-    else
-      open_file_without_flock(path, flags, &block)
-    end
+    File.open(path, flags, &block)
   end
 
   ##
   # Open a file with given flags, and protect access with flock
 
-  def self.open_file_with_flock(path, flags, &block)
+  def self.open_file_with_flock(path, &block)
+    flags = File.exist?(path) ? "r+" : "a+"
```

Comment:
This seems a race condition.
Why not `IO::RDWR | IO::APPEND | IO::CREAT` simply?

Reply:
Good point, that makes sense to me, can you create a PR?

Reply:
PR created at #7931.